{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Jupyter notebook for CHi-C data quality check\n",
    "\n",
    "This notebook is the code for comparing the CHi-C data used in paper **\"Burren, O.S. et al, Chromosome contacts in activated T cells identify autoimmune disease candidate genes, Genome Biology, 18 (165) (2017) \"** with the \n",
    "time course CHi-C data used in the paper **\"Simultaneous analysis of open chromatin, promoter interactions and gene expression in stimulated T cells implicates causal genes for rheumatoid arthritis\"** by Jing Yang, Amanda McGovern, Paul Martin, Kate Duffus, Peyman Zarrineh, Andrew P Morris, Antony Adamson, Peter Fraser, Magnus Rattray & Stephen Eyre \n",
    "\n",
    "Author : *Jing Yang*  <br />\n",
    "Date: 24-10-2019 <br />\n",
    "For any questions about the code, please drop me a line at Jing.Yang@manchester.ac.uk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load required libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n",
    "library(reshape2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_original <- read.table(\"https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/CHiC_rawdata_withscore.txt\", header=TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load bait info, to include those bait-to-bait interactions twice with either baits are being used as baits or otherEnds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "allbaitID <- read.table('https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/baitID.txt')\n",
    "data_original$baitornot <- lapply(seq(dim(data_original)[1]), function(i) sum(data_original$otherEndID[i]==allbaitID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp <- data_original[data_original$baitornot=='1',c(2,1,3:21)]\n",
    "names(tmp) <- names(data_original)\n",
    "data <- unique(rbind(data_original, tmp))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### add bait and otherEnd coordinates info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "bait_info <- read.table('https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/capture.rmap')\n",
    "names(bait_info) <- c('baitchr','baitstart','baitend','baitID')\n",
    "otherEnd_info <- bait_info\n",
    "names(otherEnd_info) <- c('otherEndchr','otherEndstart','otherEndend','otherEndID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_data <- unique(merge(merge(data, bait_info), otherEnd_info))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>277222</li>\n",
       "\t<li>27</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 277222\n",
       "\\item 27\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 277222\n",
       "2. 27\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 277222     27"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(chic_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 277222 interactions with at least one time point with Chicago score over 5"
     ]
    }
   ],
   "source": [
    "cat(\"There are\", dim(chic_data)[1], \"interactions with at least one time point with Chicago score over 5\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### retrieve merged interactions at time 0 and time 4hrs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "idxT01_1 <- chic_data$T01_s > 5\n",
    "idxT01_2 <- chic_data$T02_s > 5\n",
    "idxT01 <- idxT01_1 | idxT01_2\n",
    "\n",
    "idxT4H_1 <- chic_data$T4H1_s > 5\n",
    "idxT4H_2 <- chic_data$T4H2_s > 5\n",
    "idxT4H <- idxT4H_1 | idxT4H_2\n",
    "\n",
    "idx_combined <- idxT01 | idxT4H"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. FALSE\n",
       "2. TRUE\n",
       "3. TRUE\n",
       "4. TRUE\n",
       "5. TRUE\n",
       "6. TRUE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(idx_combined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_dataonly <- unique(chic_data[idx_combined,c('baitID','otherEndID','T01','T02','T201','T202','T1H1','T1H2','T4H1','T4H2','T24H1')])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### convert the coordinates from GRCh38 to GRCh37"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "coordinate_convert <- read.table('https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/corrdinate_convert_from38to37_withbaitID.txt', header=T)\n",
    "coordinate_convert_otherEnd <- read.table('https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/corrdinate_convert_from38to37_withotherEndID.txt', header=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "captureinfo <- coordinate_convert[,c(1,5,6,7)]\n",
    "captureinfo_otherEnd <- coordinate_convert_otherEnd[,c(1,5,6,7)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_data37 <- unique(merge(merge(chic_dataonly,captureinfo), captureinfo_otherEnd))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### add gene data associated with each bait"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "bait_gene_original <- read.table(\"https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/baits+ENSG_final_uniq_orderedbysequence.txt\", header=FALSE)\n",
    "names(bait_gene_original)[1:5] <- c('baitID','chr','baitstart','baitend','ENSG')\n",
    "bait_gene <- bait_gene_original[c(1,5)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_data37_withgene <- unique(merge(chic_data37, bait_gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_data37_withgene_sorted <- chic_data37_withgene[with(chic_data37_withgene, order(baitID,otherEndID)), ]\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load CHi-C data from Burren at al paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_data_fromGenomeBiology <- read.table('https://personalpages.manchester.ac.uk/staff/jing.yang/Data/CHiC/data/chic_counts_fromGenomeBiology.txt', header=T)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### prepare data for comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "chic_original_forcompare <- chic_data37_withgene_sorted[,c('baitID','baitchr_37','baitstart_37','baitend_37','otherEndID','otherEndchr_37','otherEndstart_37','otherEndend_37','ENSG')]\n",
    "\n",
    "chic_GenomeBiology_forcompare <- chic_data_fromGenomeBiology[,c('baitID','baitChr','baitStart','baitLength','oeID','oeChr','oeStart','oeLength','id')]\n",
    "chic_GenomeBiology_forcompare$baitEnd <- chic_GenomeBiology_forcompare$baitStart + chic_GenomeBiology_forcompare$baitLength -1\n",
    "chic_GenomeBiology_forcompare$oeEnd <- chic_GenomeBiology_forcompare$oeStart + chic_GenomeBiology_forcompare$oeLength -1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "aa_bait <- unique(chic_original_forcompare[,c('baitchr_37','baitstart_37','baitend_37')])\n",
    "bb_bait <- unique(chic_GenomeBiology_forcompare[,c('baitChr','baitStart','baitLength')])\n",
    "aa_bait$label <- 'original'\n",
    "bb_bait$label <- 'GB'\n",
    "bb_bait$baitEnd <- (bb_bait$baitStart + bb_bait$baitLength -1)\n",
    "cc_bait <- bb_bait[,c('baitChr','baitStart','baitEnd','label')]\n",
    "names(aa_bait) <- names(cc_bait)\n",
    "combined_bait <- rbind(aa_bait, cc_bait)\n",
    "combined_bait_sorted <- combined_bait[with(combined_bait, order(baitChr, baitStart)),]\n",
    "tmp <- sapply(1:(dim(combined_bait_sorted)[1]-1), function(i) combined_bait_sorted$baitStart[i+1]-combined_bait_sorted$baitStart[i])\n",
    "bait_GB_intersect <- combined_bait_sorted[(which(tmp==1)+1),c('baitChr','baitStart')]    \n",
    "bait_original_intersect <- combined_bait_sorted[which(tmp==1),c('baitChr','baitStart')]\n",
    "names(bait_original_intersect) <- c('baitchr_37','baitstart_37')\n",
    "tmp_original <- merge(chic_original_forcompare, bait_original_intersect)\n",
    "tmp_GB <- merge(chic_GenomeBiology_forcompare, bait_GB_intersect)\n",
    "chic_original_forcompare <- tmp_original\n",
    "chic_GenomeBiology_forcompare <- tmp_GB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "aa_oe <- unique(chic_original_forcompare[,c('otherEndchr_37','otherEndstart_37','otherEndend_37')])\n",
    "bb_oe <- unique(chic_GenomeBiology_forcompare[,c('oeChr','oeStart','oeLength')])\n",
    "aa_oe$label <- 'original'\n",
    "bb_oe$label <- 'GB'\n",
    "bb_oe$oeEnd <- (bb_oe$oeStart + bb_oe$oeLength -1)\n",
    "cc_oe <- bb_oe[,c('oeChr','oeStart','oeEnd','label')]\n",
    "names(aa_oe) <- names(cc_oe)\n",
    "combined_oe <- rbind(aa_oe, cc_oe)\n",
    "combined_oe_sorted <- combined_oe[with(combined_oe, order(oeChr, oeStart)),]\n",
    "tmp <- sapply(1:(dim(combined_oe_sorted)[1]-1), function(i) combined_oe_sorted$oeStart[i+1]-combined_oe_sorted$oeStart[i])\n",
    "oe_GB_intersect <- combined_oe_sorted[(which(tmp==1)+1),c('oeChr','oeStart')]    \n",
    "oe_original_intersect <- combined_oe_sorted[which(tmp==1),c('oeChr','oeStart')]\n",
    "names(oe_original_intersect) <- c('otherEndchr_37','otherEndstart_37')\n",
    "tmp_original <- merge(chic_original_forcompare, oe_original_intersect)\n",
    "tmp_GB <- merge(chic_GenomeBiology_forcompare, oe_GB_intersect)\n",
    "chic_original_forcompare <- tmp_original\n",
    "chic_GenomeBiology_forcompare <- tmp_GB"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Due to mapped to different reference genome, interactions of distance with 2 bp are deemed the same"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are  48570  and  43345  interactions in original and GB data, among them the intersected are [1] 27794\n",
      "There are  36698  and  40882  interactions in original and GB data (<5MB) , among them the intersected are [1] 26836\n",
      "There are  28028  and  34539  interactions in original and GB data (<1MB), among them the intersected are [1] 22554\n",
      "There are  21481  and  27588  interactions in original and GB data (<500k), among them the intersected are [1] 17986\n",
      "There are  9627  and  13464  interactions in original and GB data (<200k), among them the intersected are [1] 8367\n"
     ]
    }
   ],
   "source": [
    "\n",
    "chic_original_forcompare$baitpos <- 0.5*(chic_original_forcompare$baitstart_37+chic_original_forcompare$baitend_37) \n",
    "chic_original_forcompare$otherEndpos <- 0.5*(chic_original_forcompare$otherEndstart_37+chic_original_forcompare$otherEndend_37)\n",
    "chic_GenomeBiology_forcompare$baitpos <- 0.5*(chic_GenomeBiology_forcompare$baitStart+chic_GenomeBiology_forcompare$baitStart + chic_GenomeBiology_forcompare$baitLength) \n",
    "chic_GenomeBiology_forcompare$oepos <- 0.5*(chic_GenomeBiology_forcompare$oeStart+chic_GenomeBiology_forcompare$oeStart+chic_GenomeBiology_forcompare$oeLength)\n",
    "\n",
    "chic_original_forcompare$dist <- abs(chic_original_forcompare$baitpos - chic_original_forcompare$otherEndpos)\n",
    "chic_original_forcompare$pos <- 0.5*(chic_original_forcompare$baitpos + chic_original_forcompare$otherEndpos)\n",
    "\n",
    "chic_GenomeBiology_forcompare$dist <- abs(chic_GenomeBiology_forcompare$baitpos - chic_GenomeBiology_forcompare$oepos)\n",
    "chic_GenomeBiology_forcompare$pos <- 0.5*(chic_GenomeBiology_forcompare$baitpos + chic_GenomeBiology_forcompare$oepos)\n",
    "\n",
    "idx_5M_original <- chic_original_forcompare$dist<5000000\n",
    "idx_5M_GB <- chic_GenomeBiology_forcompare$dist<5000000\n",
    "\n",
    "idx_1M_original <- chic_original_forcompare$dist<1000000\n",
    "idx_1M_GB <- chic_GenomeBiology_forcompare$dist<1000000\n",
    "\n",
    "idx_500K_original <- chic_original_forcompare$dist<500000\n",
    "idx_500K_GB <- chic_GenomeBiology_forcompare$dist<500000\n",
    "\n",
    "idx_200K_original <- chic_original_forcompare$dist<200000\n",
    "idx_200K_GB <- chic_GenomeBiology_forcompare$dist<200000\n",
    "\n",
    "M <- 2\n",
    "sum_interaction_2 <- list()\n",
    "data_original <- unique(chic_original_forcompare[,c('baitchr_37','pos','baitstart_37')])\n",
    "data_GenomeBiology <- unique(chic_GenomeBiology_forcompare[,c('baitChr','pos','baitStart')])\n",
    "names(data_original) <- c('chr','pos','start')\n",
    "data_original$label <- 'original'\n",
    "names(data_GenomeBiology) <- c('chr','pos','start')\n",
    "data_GenomeBiology$label <- 'GB'\n",
    "data_combined <- rbind(data_original, data_GenomeBiology)\n",
    "data_combined_sorted <- data_combined[with(data_combined, order(chr,pos,start)),]\n",
    "sum_interaction <- 0\n",
    "for (ii in 2:(dim(data_combined_sorted)[1]-1)){\n",
    "    if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii-1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii-1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii-1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii+1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii+1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii+1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else {}\n",
    "    \n",
    "    \n",
    "}\n",
    "sum_interaction_2 <- c(sum_interaction_2, sum_interaction)\n",
    "cat('There are ', dim(data_original)[1], ' and ', dim(data_GenomeBiology)[1], ' interactions in original and GB data, among them the intersected are ')\n",
    "print(sum_interaction)\n",
    "\n",
    "\n",
    "data_original <- unique(chic_original_forcompare[idx_5M_original,c('baitchr_37','pos','baitstart_37')])\n",
    "data_GenomeBiology <- unique(chic_GenomeBiology_forcompare[idx_5M_GB,c('baitChr','pos','baitStart')])\n",
    "names(data_original) <- c('chr','pos','start')\n",
    "data_original$label <- 'original'\n",
    "names(data_GenomeBiology) <- c('chr','pos','start')\n",
    "data_GenomeBiology$label <- 'GB'\n",
    "data_combined <- rbind(data_original, data_GenomeBiology)\n",
    "data_combined_sorted <- data_combined[with(data_combined, order(chr,pos)),]\n",
    "sum_interaction <- 0\n",
    "for (ii in 2:(dim(data_combined_sorted)[1]-1)){\n",
    "    if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii-1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii-1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii-1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii+1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii+1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii+1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else {}\n",
    "    \n",
    "    \n",
    "}\n",
    "sum_interaction_2 <- c(sum_interaction_2, sum_interaction)\n",
    "cat('There are ', dim(data_original)[1], ' and ', dim(data_GenomeBiology)[1], ' interactions in original and GB data (<5MB) , among them the intersected are ')\n",
    "print(sum_interaction)\n",
    "\n",
    "data_original <- unique(chic_original_forcompare[idx_1M_original,c('baitchr_37','pos','baitstart_37')])\n",
    "data_GenomeBiology <- unique(chic_GenomeBiology_forcompare[idx_1M_GB,c('baitChr','pos','baitStart')])\n",
    "names(data_original) <- c('chr','pos','start')\n",
    "data_original$label <- 'original'\n",
    "names(data_GenomeBiology) <- c('chr','pos','start')\n",
    "data_GenomeBiology$label <- 'GB'\n",
    "data_combined <- rbind(data_original, data_GenomeBiology)\n",
    "data_combined_sorted <- data_combined[with(data_combined, order(chr,pos)),]\n",
    "sum_interaction <- 0\n",
    "for (ii in 2:(dim(data_combined_sorted)[1]-1)){\n",
    "    if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii-1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii-1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii-1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii+1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii+1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii+1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else {}\n",
    "    \n",
    "    \n",
    "}\n",
    "sum_interaction_2 <- c(sum_interaction_2, sum_interaction)\n",
    "\n",
    "cat('There are ', dim(data_original)[1], ' and ', dim(data_GenomeBiology)[1], ' interactions in original and GB data (<1MB), among them the intersected are ')\n",
    "print(sum_interaction)\n",
    "\n",
    "\n",
    "data_original <- unique(chic_original_forcompare[idx_500K_original,c('baitchr_37','pos','baitstart_37')])\n",
    "data_GenomeBiology <- unique(chic_GenomeBiology_forcompare[idx_500K_GB,c('baitChr','pos','baitStart')])\n",
    "names(data_original) <- c('chr','pos','start')\n",
    "data_original$label <- 'original'\n",
    "names(data_GenomeBiology) <- c('chr','pos','start')\n",
    "data_GenomeBiology$label <- 'GB'\n",
    "data_combined <- rbind(data_original, data_GenomeBiology)\n",
    "data_combined_sorted <- data_combined[with(data_combined, order(chr,pos)),]\n",
    "sum_interaction <- 0\n",
    "for (ii in 2:(dim(data_combined_sorted)[1]-1)){\n",
    "    if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii-1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii-1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii-1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii+1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii+1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii+1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else {}\n",
    "    \n",
    "    \n",
    "}\n",
    "sum_interaction_2 <- c(sum_interaction_2, sum_interaction)\n",
    "\n",
    "cat('There are ', dim(data_original)[1], ' and ', dim(data_GenomeBiology)[1], ' interactions in original and GB data (<500k), among them the intersected are ')\n",
    "print(sum_interaction)\n",
    "\n",
    "\n",
    "data_original <- unique(chic_original_forcompare[idx_200K_original,c('baitchr_37','pos','baitstart_37')])\n",
    "data_GenomeBiology <- unique(chic_GenomeBiology_forcompare[idx_200K_GB,c('baitChr','pos','baitStart')])\n",
    "names(data_original) <- c('chr','pos','start')\n",
    "data_original$label <- 'original'\n",
    "names(data_GenomeBiology) <- c('chr','pos','start')\n",
    "data_GenomeBiology$label <- 'GB'\n",
    "data_combined <- rbind(data_original, data_GenomeBiology)\n",
    "data_combined_sorted <- data_combined[with(data_combined, order(chr,pos)),]\n",
    "sum_interaction <- 0\n",
    "for (ii in 2:(dim(data_combined_sorted)[1]-1)){\n",
    "    #print(ii)\n",
    "    if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii-1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii-1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii-1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else if ((data_combined_sorted$label[ii] == 'original') &\n",
    "        (data_combined_sorted$label[ii] != data_combined_sorted$label[ii+1]) & \n",
    "        (abs(data_combined_sorted$pos[ii]-data_combined_sorted$pos[ii+1])<M) &\n",
    "        (abs(data_combined_sorted$start[ii]-data_combined_sorted$start[ii+1])<M)) {\n",
    "        sum_interaction <- sum_interaction +1\n",
    "    }\n",
    "    else {}\n",
    "    \n",
    "    \n",
    "}\n",
    "sum_interaction_2 <- c(sum_interaction_2, sum_interaction)\n",
    "\n",
    "cat('There are ', dim(data_original)[1], ' and ', dim(data_GenomeBiology)[1], ' interactions in original and GB data (<200k), among them the intersected are ')\n",
    "print(sum_interaction)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### plotting the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_forplot <- data.frame(Intersected=unlist(sum_interaction_2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "original_size <- c(dim(unique(chic_original_forcompare[,c('baitchr_37','pos','baitstart_37')]))[1], \n",
    "                   dim(unique(chic_original_forcompare[idx_5M_original,c('baitchr_37','pos','baitstart_37')]))[1], \n",
    "                   dim(unique(chic_original_forcompare[idx_1M_original,c('baitchr_37','pos','baitstart_37')]))[1],\n",
    "                   dim(unique(chic_original_forcompare[idx_500K_original,c('baitchr_37','pos','baitstart_37')]))[1],\n",
    "                   dim(unique(chic_original_forcompare[idx_200K_original,c('baitchr_37','pos','baitstart_37')]))[1])\n",
    "                   #sum(idx_1M_original), sum(idx_500K_original), sum(idx_200K_original))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_forplot$Remaining <- original_size - data_forplot$Intersected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_forplot$idx <- c('NONE','<5Mb','<1Mb','<500kb','<200kb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "melted_data <- melt(data_forplot, id.var=c(\"idx\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "melted_data$idx_f = factor(melted_data$idx, levels=c('NONE','<5Mb','<1Mb','<500kb','<200kb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAA+VBMVEUAAAABAQEDAwMEBAQG\nBgYICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBAfHx8iIiImJiYzMzM2NjY4ODg+Pj5BQUFE\nRERJSUlNTU1OTk5RUVFUVFRZWVlcXFxpaWlqampvb290dHR1dXV3d3d4eHh5eXl+fn6CgoKF\nhYWIiIiIoOiNjY2Npe2SkpKYmJiampqgoKCjo6Oqqqqrq6uzs7O0tLS1tbW7u7u+vr6/v7/D\nw8PExMTHx8fJycnNzc3Q0NDT09PU1NTW1tbY2NjZ2dnb29vh4eHp6ent7e3x8fHy8vL19fX4\n+Pj5+fn6+vr7+/v8/Pz+/v7///+bIurjAAAACXBIWXMAABJ0AAASdAHeZh94AAAcFklEQVR4\nnO3dCXsj6VUG0CIkIQwwYGCIhyUsgZCwDmHJsIQlQCAkhET//8fQ3fYn1a4q+8p1r33O0+22\ntZTeqrpvSyrbUncCnq07OgC8BooEARQJAmwuUkDj1haxcfEzF1u45s68Wy8e+B/P8qK6rutd\noOtf9PGsTXm60P8lNyxs5SKjs7oPnhdouGW6x8WuX/y5N/iYe/sYrl7wqUN/VJGuLbvbfvsv\nUaQPw9B1c0U6nzW3kPHynhl1fXGzC98e5nHsnx+p65013TrrV99/qW75MsWK9CRbihS1rABd\n+9j1vx6eNZcndZFmLxldpCsLjSvSkxd+egz94a6ze2z+wz+ny0nttId/Bxc79e8Qe1cdn3de\nXH/Zg+ucr3b5d2FJ63lHQbuFXOeNNHNTgwv3T5vEn7vWSvDhvpmtVP9K3XQ1W/rF7Tncph/O\nOw2veb5P7Not9QZ1fGPXwgxW9rwm3fDGJ2FHSXtX7+36XqjBhuvmJmV1tOanpV/PQW2HD/F2\nFan9PfU+7w/Z8O/4Ypf9NP+3nTc9c/7T5ZOu5p0JOrvg3vZbutB0xTYE7BaWNtwjvTWZ21mz\neUcLWtqeg+s/XHgw5P3V7/orNtgkazt1GmaykqNbmA87STr8erBlRkvfsiPWk/faNtgBo3zT\nfbPissX7G6D9/zTYOv0dObrq0uKm/7P0bqcbXGd8teUlLeY9Db/ophdvm2q4JvO3Ptqbc/EH\nq79tKR82a3tOPjlruK1Gqzaz/ee25+jGL1fvxlefbMjxwjeFGVz8/JhgeAvrUzFY/65/0vmW\n5xa6MClre3Qcoh+7m+QbfbpunGKwiy/3h6PdPjjpsrbDxw2TTda7QNdf0NKKLyxpNe846MxS\nRg9MBsMzt2Izu72bLnJu1WeWMlqTft7+2ZOF9h7b9S8zCjQ++dR/mNJ7INQ7q5tcebrjVsPM\nX3x4Cwsbdri/JtMw2GjdbOy2C6ejNb7BbngD0002XZvRp+vGgzm6/uCe9/LP4KTuctHVKepf\n6by2l20xs3nml7SYd5DgchvTpQzHoD88sys2PKkfv520cFMzS5muyXhnzS30XIHe5psLND75\nNHi8P3xk2xvzyTabLmw5zHQlu+ktrG3YwbD1d/1gy0wXOp6UXvyZPdlPPow5uOwg7PDTdWuD\nOby1/nosbpa1KVpa2272astLWsw7M0JreR4u01uhmc0/Wu/+zc5nXrzJub03OWtpO47jTFa3\nW7z+YBKGi+suixvvmx1hptu1m97C8oadOa+b3PzsQqcXn+ybma10Oa+bCzAIO/x03XgjdafT\n+M5v9BRtcLHHv6eZq06fhPYuMF707C5bWNJy3pmgM7mGW240lOP1vyx2Lf7MTYyD955zD9ek\nf9bMWgxWc5h5GmhhK/Sv11tc7zn3YIPM7dTlMINNNbpONz1xumEHm2HmoV3XjRY6zbQ+WvPT\nMlrruQHonXVVf90eFtK7527PHIf/9i526h8s7Lp2+LctbnTst7fsrn+Z7kqRRktaz9s/sX/k\ndri03mbq7eyZCw/Wexx/eJh7+n/lYPNcNudwTfpnXdZisDaXz/uZ57bnaHed5m6+O99E13XD\nNR3u3G1hhtWYufFJ2JmkvRvrelumd3p/oW1RvfWfHa3Rbh/up95a91vTDQJsL1IZGddoU6aM\nwcfSZpwLFjbjiwvoNlymosEjpEyuZUobfCRvxlsWaXatx7ss7ZZ5kv5jjUyuhsoafChzxplo\nQUVaWOvRLku8aaAORYIAigQBFAkCKBIEUCQIoEgQQJEggCJBAEWCAIoEARQJAigSBFAkCKBI\nEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIh3kJz8e+UnEUscL/XHEQtlAkQ7yoz8Z\n+VHAQn/6r2MBC2ULRTqIIr0uinQQRXpdFOkgivS6KNJB1op0eYe49d0zfDuEbr1I3co7Xmy9\nQRbZcAc5oEh6ckO27EGuFOnhnRe79jaM569Pp8vH3tmPF91SpP7i2tLaDV4+6b3xI1vYWAe5\nVqTzqHfnDo3/DM7eeo80WcjciZeesZGNdZAdRTq/p3E3+br35cbnSP3Ln99QWZGezcY6yJ57\npNP52c15vntF6p2+7R7ptHovpEhPYmMdZPdDu+n0hzy0U6QYNtZBNhWpPfg63wONPp7PPu09\n2DBs50yFarw9dB421kHyf0NWkfawsQ6SvUjukPaxtQ6SvUjso0gHUaTXRZEO8r8/HIn4Hbyf\njhf6w4CFsoUiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKBIEUCQIoEgQ\nQJEggCJBAEWCAIoEARQJAigSBFAkCKBIEECRIIAiQQBFggCKBAEUCQI8v0i/0/1nQA4oTZEg\ngCJBAEWCAIoEARQJAigSBFAkCKBIEGC1SPfvffjnNPfxgSLBtSI9fpz980iRQJEgwtWHdqtF\n+revf/3rn3xZkXjzrh1sWGrRQ5E+v3vnI0XizVu/R7pWpP/7wQ9+8Ns/o0i8edeeI60X6T3P\nkeDKQ7ulA98Of8OAb8hCAEWCAIoEARQJAigSBFAkCKBIEECRIIAiQQBFggCKBAEUCQIoEgRQ\nJAigSBDgoCL9WZRnx4cIigQBFAkCKBIEUCQIoEgQQJEggCJBAEWCAIoEARQJAigSBFAkCKBI\nEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKBIEUCQIoEgQQJEg\ngCJBAEWCAIoEARQJAigSBFAkCKBIEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEA\nRYIAigQBFAkCKBIEUCQIoEgQQJEggCJBAEWCAIoEARQJAigSBFAkCKBIEECRIIAiQQBFggCK\nBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKBIEUCQIoEgQQJEggCJBAEWCAIoEARQJ\nAigSBFAkCKBIEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKBIE\nUCQIoEgQQJEggCJBAEWCAIoEARQJAigSBFAkCHC1SPfv/tzfn+Y+PlAkuFqkd41Z+vNIkeBa\nke4VCTZYL9Jyix6K9E+ffvrpr31FkXjzVot0f+2h3T988sknv/rzisSbt16kRx7awToHGyDA\nhiI5/A3X+IYsBFAkCKBIEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQB\nFAkCKBIEUCQIoEgQQJEggCJBAEWCAIoEARQJAigSBFAkCKBIEECRIIAiQQBFggCKBAEUCQIo\nEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkj\nUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1\nKFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYci\nJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLi\nSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5E\nHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNSh\nSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqU\nOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1KFLiSNShSIkj\nUYciJY5EHYqUOBJ1KFLiSNShSIkjUYciJY5EHYqUOBJ1rBfp/v5+5eMDRbpZJOroxl913eWk\nd3VZ+fNIkW4WiToGRXrXoa7rN2m1S48U6WaRqKMbffG+Rb3T3vdlpUif373zkSLdKBJ1jIrU\nPXbpbP0e6V++9rWv/fqXFelGkahj9NBuZ5He89DuZpGoY3SwoRuc5GDDsZGow+HvxJGowzdk\nE0eijvFDu/Hh7+sU6WaRqGNysEGR8kSijvE3ZPdTpJtFoo7xN2T3U6SbRaKOyc/a7aZIN4tE\nHZ4jJY5EHYqUOBJ1+D5S4kjUoUiJI1HHqEi7H9gp0g0jUcfcTzbsW4Ii3SwSdcx8Q9bBhiyR\nqGPuG7KKlCQSdbhHShyJOjxHShyJOhy1SxyJOnwfKXEk6lCkxJGo41Kk9uqQftYuTSTqUKTE\nkajDQ7vEkahj5huyDn+LxF5+skEkAvRK03WeI4nE08wVad8SFOktRWKBVxESiQCO2olEgOm7\nUXhoJxK7jV9FaPgmslso0luKxAKHv0UigF/sE4kAfrFPJAL4xT6RCODwt0gEmCuS50gisZO3\nvhSJAHPvRrFvCYr0liKxYPx9pP0/2qBIbykSC2aK5DmSSOw1/oas50gi8QST95D1HEkk9vN9\nJJEIMPOzdjsp0luKxAJFEokAk+dIuynSW4rEgrlvyDpqJxI7KZJIBHDUTiQCKJJIBJj89LfX\n/haJ/SbPkfzQqkjs54dWRSKAX6MQiQAOf4tEAC/HJRIBHP4WiQDe+lIkAnjtb5EI4K0vRSKA\nt74UiQDe+lIkAjhqJxIBvGSxSATwkw0iEcCLn4hEAAcbRCKAVxESiQCeI4lEAEUSiQC+jyQS\nARRJJAJcinT5mVUP7URiJ0USiQAe2olEAEUSiQCKJBIBFEkkAnQzn+2jSG8pEgsGRfJDqyLx\nNA5/i0QAryIkEgH8PpJIBHDUTiQCjIq0/2XtFOlNRWKBd6MQiQAzL37iYINI7OVF9EUigHsk\nkQjgOZJIBHDUTiQC+D6SSARQJJEIoEgiEUCRRCKAIolEAG/rIhIBFEkkAnhbF5EI4N0oRCKA\nIolEAEftRCKAIolEgMlPf+/uliK9pUgsmDxH8vtIIrHf+Ddku/FpVynSW4rEgpkiuUcSib0c\n/haJAH7VXKTikXJw+Fuk4pFyUCSRikfKwYufiFQ8Ug6eI4lUPFIOXiBSpOKRcvCSxSIVj5TD\nzC/2eWgnUqVIOXgPWZGKR8rhoCL9e5T9N70k4YiIVMdB30dSJJFeF0VqEo6ISHXMfR/JQ7sk\nIyJSHQf99LciifS6zH0faR9FutmIiFTHQa+0qkgivS7rz5Hu7+9XPj5QpJuNiEh1rD5HeleX\nlT+PFOlmIyJSHdeeI60X6X++973v/dYXFOlGIyJSHddeRH+9SJ/fvfORIt1oRESq48rh7/d9\nWSnSf3zzm9/8jS8p0o1GRKQ61ov0eGTBc6RjRkSkOlYPeD+0RZGOGhGR6rhSpPuHx3YOfx8y\nIiLV4UeEmoQjIlIditQkHBGR6pgpzUv8qrkiifS6zLXGPVKSERGpjrl7JEVKMiIi1eE5UpNw\nRESqY65I+5agSDcbEZHq8JoNTcIREakORWoSjohIdXhduybhiIhUhyI1CUdEpDqmpXGwIc2I\niFTHuDX7X/9EkW42IiLVMXnxk91LUKSbjYhIdXg5ribhiIhUR686T6qRIt1wRESqw1G7JuGI\niFSHIjUJR0SkOvxkQ5NwRESqQ5GahCMiUh2K1CQcEZHqUKQm4YiIVIciNQlHRKQ6FKlJOCIi\n1aFITcIREakORWoSjohIdShSk3BERKpDkZqEIyJSHYrUJBwRkepQpCbhiIhUhyI1CUdEpDoU\nqUk4IiLVoUhNwhERqQ5FahKOiEh1KFKTcEREqkORmoQjIlIditQkHBGR6lCkJuGIiFSHIjUJ\nR0SkOhSpSTgiItWhSE3CERGpDkVqEo6ISHUoUpNwRESqQ5GahCMiUh2K1CQcEZHqUKQm4YiI\nVIciNQlHRKQ6FKlJOCIi1aFITcIREakORWoSjohIdShSk3BERKpDkZqEIyJSHYrUJBwRkepQ\npCbhiIhUhyI1CUdEpDoUqUk4IiLVoUhNwhERqQ5FahKOiEh1KFKTcEREqkORmoQjIlIditQk\nHBGR6lCkJuGIiFSHIjUJR0SkOhSpSTgiItWhSE3CERGpDkVqEo6ISHUoUpNwRESqQ5GahCMi\nUh2K1CQcEZHqUKQm4YiIVIciNQlHRKQ6FKlJOCIi1aFITcIREakORWoSjohIdShSk3BERKpD\nkZqEIyJSHYrUJBwRkepQpCbhiIhUhyI1CUdEpDoUqUk4IiLVoUhNwhERqQ5FahKOiEh1KFKT\ncEREqkORmoQjIlIditQkHBGR6lCkJuGIiFSHIjUJR0SkOhSpSTgiItWhSE3CERGpDkVqEo6I\nSHUoUpNwRESqQ5GahCMiUh2K1CQcEZHqUKQm4YiIVIciNQlHRKQ6FKlJOCIi1aFITcIREakO\nRWoSjohIdShSk3BERKpDkZqEIyJSHYrUJBwRkepQpCbhiIhUhyI1CUdEpDoUqUk4IiLVoUhN\nwhERqQ5FahKOiEh1KFKTcEREqkORmoQjIlIditQkHBGR6lCkJuGIiFSHIjUJR0SkOhSpSTgi\nItWhSE3CERGpDkVqEo6ISHUoUpNwRESqQ5GahCMiUh2K1CQcEZHqUKQm4YiIVIciNQlHRKQ6\nFKlJOCIi1aFITcIREakORWoSjohIdShSk3BERKpDkZqEIyJSHVeKdH//4cP8xweKdLMREamO\n9SJ9KMy7v7N/HinSzUZEpDo23COtFOm/vvOd73z6s4p0oxERqY7nFenzu3c+UqQbjYhIdTyv\nSN//7LPPvvpFRbrRiIhUx/OK9N5reY6UMFLCqU0YKQdFShwp4dQmjJSDw9+JIyWc2oSRcvAN\n2cSREk6tSAsUKXGkHCMi0haKlDhSjhERaQtFShwpx4iItIUiJY6UY0RE2kKREkfKMSIibaFI\niSPlGBGRtlCkxJFyjIhIWyhS4kg5RkSkLRQpcaQcIyLSFoqUOFKOERFpC0VKHCnHiIi0hSIl\njpRjRETaQpESR8oxIiJtoUiJI+UYEZG2UKTEkXKMiEhbKFLiSDlGRKQtFClxpBwjItIWipQ4\nUo4REWkLRUocKceIiLSFIiWOlGNERNpCkRJHyjEiIm2hSIkj5RgRkbZQpMSRcoyISFsoUuJI\nOUZEpC0UKXGkHCMi0haKlDhSjhERaQtFShwpx4iItIUiJY6UY0RE2kKREkfKMSIibaFIiSPl\nGBGRtlCkxJFyjIhIWyhS4kg5RkSkLRQpcaQcIyLSFoqUOFKOERFpC0VKHCnHiIi0hSIljpRj\nRETaQpESR8oxIiJtoUiJI+UYEZG2UKTEkXKMiEhbKFLiSDlGRKQtFClxpBwjItIWipQ4Uo4R\nEWkLRUocKceIiLSFIiWOlGNERNpCkRJHyjEiIm2hSIkj5RgRkbZQpMSRcoyISFsoUuJIOUZE\npC0UKXGkHCMi0haKlDhSjhERaQtFShwpx4iItIUiJY6UY0RE2kKREkfKMSIibaFIiSPlGBGR\ntlAkkfbIMbUJIymSSHvkmNqEkRRJpD1yTG3CSIok0h45pjZhJEUSaY8cU5swkiKJtEeOqU0Y\nSZFE2iPH1CaMpEgi7ZFjahNGUiSR9sgxtQkjKZJIe+SY2oSRFEmkPXJMbcJIiiTSHjmmNmEk\nRRJpjxxTmzCSIom0R46pTRhJkUTaI8fUJoykSCLtkWNqE0ZSJJH2yDG1CSMpkkh75JjahJEU\nSaQ9ckxtwkiKJNIeOaY2YSRFEmmPHFObMJIiibRHjqlNGEmRRNojx9QmjKRIIu2RY2oTRlIk\nkfbIMbUJIymSSHvkmNqEkRRJpOKRFCnX/hCpaCRFyrU/RCoaSZFy7Q+RikZSpFz7Q6SikRQp\n1/4QqWgkRcq1P0QqGkmRcu0PkYpGUqRc+0OkopEUKdf+EKloJEXKtT9EKhpJkXLtD5GKRlKk\nXPtDpKKRFCnX/hCpaCRFyrU/RCoaSZFy7Q+RikZSpFz7Q6SikRQp1/4QqWgkRcq1P0QqGkmR\ncu0PkYpGUqRc+0OkopEUKdf+EKloJEXKtT9EKhpJkXLtD5GKRlKkXPtDpKKRFCnX/hCpaCRF\nyrU/RCoaSZFy7Q+RikZSpFz7Q6SikRQp1/4QqWgkRcq1P0QS6RkUSSSRAiIpkkgiBURSJJFE\nCoikSCKJFBBJkUQSKSCSIokkUkAkRRJJpIBIiiSSSAGRFEkkkQIiKZJIIgVEUiSRRAqI9LQi\n3d/fnz9XJJFEelqR3tXo0iRFEkkkRRJJpIhIzyvS9z/77LOvflGRRHrzkZ5XpM/v3vnoCUWC\n1+V5Rfrv7373u7/5BUXizTvoORK8Lgcd/obX5aBvyMLrokgQQJEggCJBAEWCAIoEARQJAigS\nBFAkCKBIEECRIIAiQQBFggCKBAEUCQIoEgRQJAigSBBAkSCAIkEARYIAigQBFAkCKBIEUCQI\noEgQQJEgQESR/uBb8Cb9RWCRPv+V7iY++vgrt1nw033l44+OjjD2cx//wtERxr788S8eHWHs\nSx//0m0W/MuBRTr9/bdv4g/vvnGbBT/dN+7+6OgIY3989/tHRxj71t3vHR1h7E/vfvc2C/67\nyCLdyLfv/vboCGN/c/fXR0cY+87dXx0dYewf7/7y6Ahj/3z357e+CUXaQZG2UKRcFGkLRdri\nTRcJClEkCKBIEECRIECmIn14q/THD71P33/Vexf1Y5Ldf8jx+MXp4DgP3odIE+lxH91fdt9D\noCNj3Q9m6caRchXp/jwe571y/IycLnug7ZLjE523TpJI5xnt/zkdWqRRmBtHylik3j1RhiL1\n9sBlao/NdL7DzhGpP6OTP8fEWs5zo0i5inS6/MfxuNLHP7R7nNf2sT3IO/YxyyDY0ZH6W2g8\nrwfFGjzkfZFINYp0bKhzuA95TsdO7ak/JSkijbZQb2pPRxVptIVeIlKyIl0e7t/fD78+NtWp\ntysOL9LgHilBpNEWSnOP1P8P+a3dI536q5qjSINdMRiYQyNdjtodH2lSofGfoyKtZIuWrUjt\nQ67D3+d+n6f26ECnaZGOvuOeOdZ8f+Sumz8if6tImYoEZSkSBFAkCKBIEECRIIAiQQBFggCK\nBAEU6WU8vJ7g46fPXdTiJxzGLngZH2a9e/w4OvkJixou6HlFUsMINuLLeJjWycw+dYgnjXwG\nRYpgI76M3sO6dufUPT7ga1+8P+Py8O+xeG3K2xU/XObxeufLn8/o3UWdm9u7Lzz1r3S5g5xk\n6N8w29hcL2NUpP7f3he9ZgwuMS7SpR3d9Mrnc069lgyv0b+RSYb+TbCVrfUyJn0YHXiYadWG\nIo0uf17c4Jy1Bc/e7CUem9lgL2P+od1lbqej3U6/dGN8PxVbpF4GD+32s7lexrhIp9O0OzNF\nGizgtkUa3aQm7WNrvYzRRLcvlos0eZqyMPy3KtLwxrnG1noZ7fHTqQ3v+d7lciBt0I7+pc5L\n6A9/N+1H/zHh4IRu4cLnf0/DQwyDG2YLW+tljH+yoX11Lthw9CeHv5fuiBbvkXqL7y/vNKrL\nOND5JD3ax+Z6jbTgxdnir5EivThb/DVSpBdni0MARYIAigQBFAkCKBIEUCQIoEgQQJEggCJB\ngP8H24SBJAGezmMAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      },
      "text/plain": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "ggplot(data=melted_data, aes(fill=variable, x=idx_f, y=value)) +\n",
    "geom_bar(stat=\"identity\", width=0.5, alpha=0.6) + xlab('Distance within promoters') + ylab('Number of Interactions') + #geom_text(aes(label=scales::percent(value)), position = position_stack(vjust = .5))+\n",
    "ggtitle('Intersected and remainning interactions when comparing CHi-C data from our study to data in Burren et all (Supplementary Fig. 1a)') +\n",
    "theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), \n",
    "    panel.background=element_blank(), axis.line=element_line('black'), legend.position='top',    legend.text = element_text(size=6, face='bold'),\n",
    "      legend.key.height =unit(0.4,'cm'), legend.key.width = unit(0.4,'cm'),\n",
    "     axis.text.x=element_text(size=6, face='bold',hjust=0.5), axis.text.y=element_text(size=6, face='bold'),axis.title=element_text(size=8, face='bold'), \n",
    "      strip.text.x = element_text(size = 6), strip.text.y = element_text(size=6),plot.title = element_text(size=8, hjust=0.5, vjust=2.5)) + \n",
    "      scale_x_discrete(breaks=c('NONE','<5Mb','<1Mb','<500kb','<200kb'), labels=c('NONE','<5Mb','<1Mb','<500kb','<200kb')) +\n",
    "          #scale_y_continuous(labels=scales::percent) + scale_fill_brewer() + labs(fill = \" \") + scale_color_grey() + \n",
    "scale_fill_manual(\"legend\", values = c(\"Intersected\" = \"royalblue\", \"Remaining\" = \"grey\")) + guides(fill=guide_legend(' ')) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
